src/runtime/storage-proxy.ts

```diff
@@ -553,6 +565,64 @@ export class BigCollectionProxy extends StorageProxy implements BigCollectionSto
   }
 }
 
+export class DummyProxy extends StorageProxy implements CollectionStore, BigCollectionStore, SingletonStore {
```

Comment:
A few thoughts:
- We could add this class (and the `newDummyProxy` method) in a separate smaller PR. Defer to @piotrswigon on whether that's warranted. A documentation comment on this class could also be a good place for the explanation of its purpose/existence (re: my earlier comment on this PR).
- We might want to fully name this (and related methods) `DummyStorageProxy` to disambiguate from other dummy proxies we could end up with down the line, again look to @piotrswigon for thoughts.
- We should describe what the functional behavior of this class is for the reader, for example, "silently absorbs and throws away all changes, aiming to surface no logging, warnings, or otherwise affect any visible behavior or persistent changes."

Reply:
All good comments from Walter. I don't think we need a separate PR for it, but adding some documentation would be great. Also naming, wdyt about a `NoOp` as opposed to `Dummy`? It may be just me, but `NoOp` sounds more descriptive - i.e. that it does nothing.

Once you land the other PR and update this one please add Myk as a reviewer, he may have thoughts as he's as close to the owner of this code as we have.

Reply:
Added the description, also changed the class name into `NoOpStorageProxy` :D